src/lib/cache/cacheManager.ts

```diff
@@ -0,0 +1,356 @@
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { name } = category;
+    Logger.debug(`Clearing cache for command category "${name}"`);
+    await inMemoryCache.del(`${memoryCachePrefixCategory}:${name.toLowerCase()}`);
+}
+
+export async function loadSinglePrefixCommandCategoryToCache(category: IPrefixCommandCategory) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { name } = category;
+    Logger.debug(`Loading category ${name} to cache`);
+    await inMemoryCache.set(`${memoryCachePrefixCategory}:${name.toLowerCase()}`, category.toObject());
+}
+
+export async function loadAllPrefixCommandCategoriesToCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    const PrefixCommandCategories = await PrefixCommandCategory.find();
+    for (const category of PrefixCommandCategories) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandCategoryToCache(category);
+    }
+}
+
+export async function refreshSinglePrefixCommandCategoryCache(oldCategory: IPrefixCommandCategory, newCategory: IPrefixCommandCategory) {
+    await clearSinglePrefixCommandCategoryCache(oldCategory);
+    await loadSinglePrefixCommandCategoryToCache(newCategory);
+}
+
+export async function refreshAllPrefixCommandCategoriesCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    // Step 1: Get all catagories from the database
+    const prefixCommandCategories = await PrefixCommandCategory.find();
+    // Step 2: Get all categories from the cache
+    const cacheKeys = await inMemoryCache.store.keys();
+    // Step 3: Loop over cached categories
+    for (const key of cacheKeys) {
+        if (key.startsWith(`${memoryCachePrefixCategory}:`)) {
+            const categoryName = key.split(':')[1];
+            // Step 3.a: Check if cached category exists in the database list
+            let found = false;
+            for (const dbCategory of prefixCommandCategories) {
+                const { name: dbCategoryName } = dbCategory;
+                if (dbCategoryName.toLowerCase() === categoryName.toLowerCase()) {
+                    found = true;
+                    break;
+                }
+            }
+            // Step 3.b: If not found, remove from cache
+            if (!found) {
+                Logger.debug(`Removing category ${categoryName} from cache`);
+                // eslint-disable-next-line no-await-in-loop
+                await inMemoryCache.del(key);
+            }
+        }
+    }
+    // Step 4: Loop over database categories and update cache
+    for (const dbCategory of prefixCommandCategories) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandCategoryToCache(dbCategory);
+    }
```

Comment:
```suggestion
    await Promise.all(prefixCommandCategories.map((dbCategory) => loadSinglePrefixCommandCategoryToCache(dbCategory)));
```
Awaiting *all* the promises...